shared/src/commonMain/kotlin/io/customer/shared/tracking/api/HttpClientBuilder.kt

```diff
@@ -0,0 +1,79 @@
+package io.customer.shared.tracking.api
+
+import io.customer.shared.device.UserAgentStore
+import io.customer.shared.sdk.config.NetworkConfig
+import io.customer.shared.sdk.meta.Region
+import io.customer.shared.sdk.meta.Workspace
+import io.customer.shared.util.LogLevel
+import io.customer.shared.util.Logger
+import io.ktor.util.*
+
+/**
+ * Helper class to provide http client params from a single place.
+ */
+internal interface HttpClientBuilder {
```

Comment:
Can we use a Kotlin data class or something besides a Builder pattern? A builder seems overkill and heavy boilerplate here.

Reply:
Data classes are something used for holding data, and should not ideally be responsible for manipulating it or holding business logic. Or in simpler words, I would never want data class to have unit tests for it. But here, `HttpClientBuilder` determines the correct `baseURL` and generates `authHeader`, which, at some point, we'll be covering in test cases. If we replace this with data class, who do you think should take responsibility for these decisions with the graph class being its only caller?

And just wanted to add, this is not really a builder class. I was running out of names and had to give this 😅 But if you see its implementation, you'll observe that it provides the values required by our http client so it can be built easily without if-else or mappings.

Reply:
You have a good point in regards to logic existing in data classes. I would agree a data class wouldn't be a good use to hold logic like `HttpClientBuilderImpl` has. 

Now looking over the `HttpClientBuilder` file again, I am not sure why I originally made this comment because indeed, the implementation is not a builder design pattern? I must have quickly read the name, saw "Builder", and then thought, "Since we're using Kotlin, can we not use the Java Builder design pattern for constructing a class?" So, disregard my original question.